edgelet/edgelet-http/src/lib.rs

```diff
@@ -339,7 +338,7 @@ impl HyperExt for Http {
                         )
                     })?;
 
-                let cert = match cert_manager {
+                let cert = match tls_params.as_ref().map(|params| params.cert_manager) {
```

Comment:
If you're going to use a combinator, you might as well use one for the whole thing instead of also using `match`.

```
tls_params.as_ref().and_then(|params| params.cert_manager.get_pkcs12_certificate).ok_or(ErrorKind::CertificateCreationEror)?;
```